pyomo/core/plugins/transform/bounds_to_vars.py

```diff
@@ -0,0 +1,84 @@
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the transformation."""
+        super(ConstraintToVarBoundTransform, self).__init__(*args, **kwargs)
+
+    def _create_using(self, model):
+        """Create new model, applying transformation."""
+        m = model.clone()
+        self._apply_to(m)
+        return m
+
+    def _apply_to(self, model):
+        """Apply the transformation to the given model."""
+        m = model
+
+        for constr in m.component_data_objects(ctype=Constraint,
+                                               active=True,
+                                               descend_into=True):
+            # Check if the constraint is k * x + c1 <= c2 or c2 <= k * x + c1
+            if constr.body.polynomial_degree() == 1:
+                repn = generate_canonical_repn(constr.body)
+                if repn.variables is not None and len(repn.variables) == 1:
+                    var = repn.variables[0]
+                    const = repn.constant if repn.constant is not None else 0
+                    coef = float(repn.linear[0])
+                    if coef == 0:
+                        # This can happen when a one element of a bilinear term
+                        # is fixed to zero. Obviously, do not divide by zero.
+                        pass
+                    else:
+                        if constr.upper is not None:
+                            newbound = (value(constr.upper) - const) / coef
+                            if coef > 0:
+                                var.setub(min(var.ub, newbound)
+                                          if var.ub is not None
+                                          else newbound)
+                            elif coef < 0:
+                                var.setlb(max(var.lb, newbound)
+                                          if var.lb is not None
+                                          else newbound)
+                        if constr.lower is not None:
+                            newbound = (value(constr.lower) - const) / coef
+                            if coef > 0:
+                                var.setlb(max(var.lb, newbound)
+                                          if var.lb is not None
+                                          else newbound)
+                            elif coef < 0:
+                                var.setub(min(var.ub, newbound)
+                                          if var.ub is not None
+                                          else newbound)
+                    constr.deactivate()
+                    # Sometimes deactivating the constraint will remove the
+                    # variable from all active constraints, so that it won't be
+                    # updated during the optimization. Therefore, we need to
+                    # shift the value of var as necessary in order to keep it
+                    # within the constraints.
```

Comment:
Several comments:
1) Unclear exactly what this comment is saying. Are you indicating that it is possible that the variable is only in constr, therefore, when we deactivate it, the variable does not appear in any constraints at all?

2) Also, why is it necessary to shift the value? It is possible to have variable values that are outside the bounds. Is this just "cleanup"?

3) If I understand correctly, then "keep it within the constraints" should read "keep it within the bounds".

Reply:
1. Yep.
2. Yes, this is a form of cleanup.
3. Yeah, I can change that.

Reply:
On further thought, with 2: It makes sense to shift the value because we are removing a constraint from the system and shifting its "effect" directly onto the variable. In order for post-solve variable values to remain as they would be had we not performed the transformation, we need to move the variable value if it falls outside of the new bound.